File: 2015/c/day01.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "input.h"

int main(void)
{
	int nLines; char **lines; char *line;
	int floor;
	int i;

	inputLines(&nLines, &lines, "../inputs/day01input.txt");
	line = lines[0];

	/* part 1 */
	floor = 0;

	for (i = 0; i < strlen(line); i++) {
		if (line[i] == '(') {
			floor++;
		} else if (line[i] == ')') {
			floor--;
		}
	}

	printf("%d\n", floor);

	/* part 2 */
	floor = 1;

	for (i = 0; i < strlen(line); i++) {
		if (line[i] == '(') {
			floor++;
		} else if (line[i]) {
			floor--;
		}

		if (floor == -1) {
			printf("%d\n", i);
			break;
		}
	}

	return (EXIT_SUCCESS);
}

File: 2015/python/day01.py
#!/usr/bin/env python3

"""
Day 1: Not Quite Lisp
https://adventofcode.com/2015/day/1
"""

from inputlines import inputLines


def main():
    lines = inputLines('../inputs/day01input.txt')
    line = lines[0]

    # part 1
    print(line.count('(') - line.count(')'))

    # part 2
    floor = 0
    for (i, c) in enumerate(line, 1):
        if c == '(':
            floor += 1
        elif c == ')':
            floor -= 1

        if floor == -1:
            break

    print(i)


if __name__ == '__main__':
    main()
